Lib/email/_header_value_parser.py

```diff
@@ -1101,13 +1101,13 @@ def get_unstructured(value):
             token, value = get_fws(value)
             unstructured.append(token)
             continue
+        encoded_format_invalid = False
         if value.startswith('=?'):
             try:
                 token, value = get_encoded_word(value)
-            except errors.HeaderParseError:
-                # XXX: Need to figure out how to register defects when
-                # appropriate here.
-                pass
+            except errors.HeaderParseError as e:
+                if "encoded word format invalid" in str(e):
```

Comment:
I don't like doing string matching on exception message, if this is really something that needs to be caught and handled differently than generic `HeaderParseError`, it should be an exception of it's own, perhaps subclassed from `HeaderParseError`.

Reply:
Done, created `InvalidEwError`.